Scripts/FilterByList/FilterByList.py

```diff
@@ -2,64 +2,97 @@
 from CommonServerPython import *
 from CommonServerUserPython import *
 
-lst = demisto.executeCommand("getList", {"listName": demisto.args()["listname"]})
 
-if isError(lst[0]):
-    demisto.results(lst)
-    sys.exit(0)
+def empty_list_context(lst, items, list_name):
+    not_white_listed = []  # type: list
+
+    for item in items:
+        not_white_listed.append(item)
+
+    ec = {'List.In': [], 'List.NotIn': not_white_listed}
+
+    return {
+        'ContentsFormat': formats['text'],
+        'Type': entryTypes['note'],
+        'HumanReadable': 'The list ' + list_name + ' is empty',
+        'Contents': None,
+        'EntryContext': ec
+    }
 
-notwhitelisted = []  # type: list
-whitlisted = []  # type: list
-hr = ""
-ignore_case = demisto.args().get("ignorecase", "").lower() == "yes"
 
-items = demisto.args().get("values", "")
-if not isinstance(items, list):
-    items = items.split(",")
+def build_filtered_data(lst, items, ignore_case, match_exact, regex_ignore_case_flag):
+    not_white_listed = []  # type: list
+    white_listed = []  # type: list
+    human_readable = ''
 
-if not lst[0]["Contents"]:
+    # fill whitelisted array with all the the values that match the regex items in listname argument
     for item in items:
-        notwhitelisted.append(item)
-
-    ec = {"List.In": whitlisted, "List.NotIn": notwhitelisted}
-
-    demisto.results(
-        {
-            "ContentsFormat": formats["text"],
-            "Type": entryTypes["note"],
-            "Contents": 'The list ' + demisto.args()["listname"] + ' is empty',
-            "EntryContext": ec
-        }
-    )
-    sys.exit(0)
-
-lst = lst[0]["Contents"].split(",")
-search_flag = re.IGNORECASE if ignore_case else 0
-
-# fill whitelisted array with all the the values that match the regex items in listname argument
-for item in items:
-    for list_item in lst:
-        if list_item and re.search(item, list_item, search_flag):
-            hr += item + " is in the list\n"
-            whitlisted.append(item)
-
-# fill notwhitelisted array with all the the values that not in whitelisted
-for item in items:
-    if item not in whitlisted:
-        hr += item + " is not part of the list\n"
-        notwhitelisted.append(item)
-
-ec = {"List.In": whitlisted, "List.NotIn": notwhitelisted}
-contents = {"inList": whitlisted, "notInList": notwhitelisted}
-demisto.results(
-    {
-        "ContentsFormat": formats["json"],
-        "Type": entryTypes["note"],
-        "Contents": contents,
-        "HumanReadable": hr,
-        "HumanReadableFormat": formats["markdown"],
-        "EntryContext": ec
+        if match_exact:
+            if ignore_case:
+                if not item.lower() in [list_item.lower().strip() for list_item in lst]:
```

Comment:
calculate this outside the loop: `[list_item.lower().strip() for list_item in lst]`
better performance.